rust/src/arrow/kernels.rs

```diff
@@ -40,9 +40,21 @@ where
         .max_by(|(_, x), (_, y)| match (x, y) {
             (None, _) => Ordering::Less,
             (Some(_), None) => Ordering::Greater,
-            (Some(vx), Some(vy)) => vx.partial_cmp(vy).unwrap(),
+            (Some(vx), Some(vy)) => {
+                if let Some(ordering) = vx.partial_cmp(vy) {
+                    ordering
+                } else {
+                    Ordering::Greater
+                }
+            }
+        })
+        .and_then(|(idx, val)| {
+            // If the value is NaN, return None
+            if let Some(val) = val {
```

Comment:
Is it easier / faster to just run a `for loop` here?

Reply:
for loop over what?

Reply:
Just a loop to run `argmin` together.

```
let mut min_idx = None;
let min_value = T::Max;
for v in array {
   if let Some(val) = v {
       if let Some(o) = vx.partial_cmp(&min_val) {
          ...
       }
   }
}
```

Reply:
That would be a lot simpler, but there isn't any `T::Max`. Can't find anything like that in the arrow crate so far.

Reply:
We could probally either make a new trait `MaxValue`

and change the signature to

```
T: ArrowNumericType + MaxValue
```

Or just limit `argmin` to floats?

```
T:  ArrowNumricType + num_traits::float::Float
```

Reply:
Just did the first option. LMK what you think.